state/protocol/badger/snapshot.go

```diff
@@ -323,13 +323,24 @@ func (s *Snapshot) Params() protocol.GlobalParams {
 	return s.state.Params()
 }
 
-// ProtocolState returns the dynamic protocol state that the Head block commits to. The
-// compliance layer guarantees that only valid blocks are appended to the protocol state.
+// EpochProtocolState returns the epoch part of dynamic protocol state that the Head block commits to.
+// The compliance layer guarantees that only valid blocks are appended to the protocol state.
```

Comment:
```suggestion
// The compliance layer guarantees that only valid blocks are appended to the protocol state.
// TODO: Conceptually the epoch protocol state is part of the overall Dynamic Protocol State
//   (`ProtocolState` below), but it is separate here for historical reasons. Eventually we should
//   move the epoch protocol state data into the protocol state KV Store.
```

Reply:
regarding [your suggestion](https://github.com/onflow/flow-go/pull/5650#discussion_r1561525307) @jordanschalm:
* As I understand your suggestion, you are commenting on a future revision of the implementation. It is _not_ a suggestion to change the API ([`Snapshot` interface](https://github.com/onflow/flow-go/blob/43c3ceb44c015c8c652ebe5ed43b7d3493971801/state/protocol/snapshot.go#L138-L142)), right? 
* Therefore, I would suggest to move your suggested `TODO` comment into `EpochProtocolState()`'s _body_ (before [line 332](https://github.com/onflow/flow-go/blob/43c3ceb44c015c8c652ebe5ed43b7d3493971801/state/protocol/badger/snapshot.go#L332))

Reply:
📝 added item to [issue 5666](https://github.com/onflow/flow-go/issues/5666)